### ISSUE TITLE
Request updates to GitHub dataset export 

### ISSUE THREAD

**Issue by sterkenburgsara** [department-of-veterans-affairs/va.gov-team#55104]
## User Story
_As a governance team member, I an updated dataset from Github, so that the data values that we have available to us for manipulation in our Domo dashboard accurately reflect the fullest state of possibilities to date and do not cause errors (such as double counting) in our queries.

Assignee: @it-harrison 
Peer Reviewer: @sterkenburgsara 

## Description
While fulfilling ticket # within this epic, @it-harrison & @shiragoodman realized there was double counting (sometimes triple or quadruple) on ticket counts due to the nature of how practice-area labels are being pulled in. We would like to request a modification of how that data is being exported from GH using BigQuery to eliminate this problem and accurately deliver data visualizations in our dashboard.

Since we have data expansion planned for our next Phase 3 Domo epic, we plan to also request those additional labels from GH now, with this request, to minimize the number of times we return to the data & analytics team with changes. 

## Impacted Artifacts
* GH dataset export in BigQuery (that feed into Domo)
* Metadata mapping instructions

## Tasks
- [ ] @it-harrison and/or @sterkenburgsara attend Analytics office hours to make request for dataset changes (and get a sense of how long this request will take)
- [ ] Once changes are made, review dataset and ensure the changes are correct and meet our needs
- [ ] Re-write queries that are populating current Domo cards based on revised dataset

## Peer Review
_To be completed by peer reviewer_
- [ ] User story and acceptance criteria are met

## Acceptance Criteria
- [ ] Domo dashboard pulls a revised & updated dataset from GH to populate the Collab Cycle dashboard
- [ ] Domo cards accurately reflect correct/consistent ticket counts

### Refinement
- [ ] Ticket has user story, description, tasks, and acceptance criteria
- [ ] Ticket has [additional appropriate labels](https://vfs.atlassian.net/wiki/spaces/PPST/pages/2220359751/WIP+Tech+Debt+Ticket+Organization), if applicable
- [ ] Ticket has been sized 
- [ ] Once all above items are checked, add the 'Ready' label.
### Planning
_If this ticket is picked up from the Backlog mid-sprint, connect with Shira to ensure the below items are completed correctly_
- [ ] Ticket has been discussed as a team at Planning
- [ ] Ticket has been assigned to appropriate initiative or quarterly epic
- [ ] Ticket has been assigned to a Sprint
- [ ] Ticket has been moved to the Planned pipeline in the Governance team board
- [ ] Ticket as an assignee and peer reviewer

**Comment by sterkenburgsara** [comment]
@shiragoodman ticket drafted - let us know if you have any edits. // cc @it-harrison

**Comment by shiragoodman** [comment]
thanks @sterkenburgsara ! I added this ticket as a dependency for ticket #54676, and I have the following comments:

1. Create 4 new fields from the existing **Practice area** field: 
  a. **IA** denoted by the `ia` label begin present on the ticket
  b. **Content** denoted by the `content` label begin present on the ticket
  c. **Design** denoted by the `design` label begin present on the ticket
  d. **Accessibility** denoted by the `508/Accessibility` label begin present on the ticket
 
2. Create 1 new field for **Launch Blocking** denoted by the label `launch-blocking` being present on the ticket.
3. Create 1 new field for **Accessibility Defect Level** denoted by 1 of the following labels `508-defect-0`, `508-defect-1`, `508-defect-2`, `508-defect-3`, and `508-defect-4` being present on the ticket

**Comment by shiragoodman** [comment]
also, check out the [Anatomy of a Staging Review ticket ](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/anatomy-of-a-staging-review-issue-ticket#AnatomyofaStagingReviewIssueticket-Accessibilitylabels) guidance to better understand what our labels are and how they're applied.

**Comment by sterkenburgsara** [comment]
Got it - I have most of that in the ticket area "description" above (with less clarity, to be sure). I'll update with your changes & make sure to review/reference the Anatomy of a Staging review guidance.

**Comment by shiragoodman** [comment]
i had started writing this out when you were still making changes to the ticket, so thought I'd like you work on yours and just add my concerns as comments. As long as you understand, sounds good to me!

**Comment by sterkenburgsara** [comment]
Question: should we also request the touchpoint labels? (for instance, I think it would be interesting down the road to see how many teams with launch-blocking issues skipped a touchpoint)? // cc @it-harrison

**Comment by shiragoodman** [comment]
assuming this isn't going to be complete by EOD so I added the next sprint. thanks for taking this on @sterkenburgsara and @it-harrison !

**Comment by sterkenburgsara** [comment]
Ian and I attended Analytics office hours on 3/14/23 to discuss these requests. We filled out [this ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/55145) to start the process and the Analytics team communicated that they will try to get it into their next sprint. We are an edge case, so they may come back with follow-up questions. // cc @it-harrison @shiragoodman

**Comment by sterkenburgsara** [comment]
@it-harrison @shiragoodman update: attended analytics office hours today 3/21/23 to get update on the dataset modifications requested last week. 

Phillip Dorsey is doing this for us and was holding the office hours, yay! His ETA is end of this week on the first 3 bullets outlined above. The last bullet point re: text-mining our body text we agreed to hold off on since it is more complex/unknown, and not crucial per our existing Phase 2 initiative tickets. Phillip said this is probably doable, but will need to be a different more experimental ticket in their process, which he can talk to Michelle about queuing up soon.

**Comment by sterkenburgsara** [comment]
@shiragoodman Update: Phillip has implemented the changes we requested (per comment above) and either @it-harrison or myself will attend Analytics office hours today (3/23) to go over the changes and hopefully sign-off. If everything looks good, he'll push the updates to our Domo instance.

**Comment by sterkenburgsara** [comment]
Status update: Ian approved dataset in Analytics office hours yesterday (3/23/23), with caveat that the third request outlined in this ticket (508 accessibility defect level) is not exactly as we had intended. @dorseypa1 is going to fix that field by the end of this sprint (next Wednesday) but for now will hand off dataset and the team will get it into Domo (estimation in the meeting was EOD Friday 3/24/23). That will allow Ian access to it in the meantime. Does that all sound correct, @it-harrison ? cc // @shiragoodman

**Comment by it-harrison** [comment]
> Status update: Ian approved dataset in Analytics office hours yesterday (3/23/23), with caveat that the third request outlined in this ticket (508 accessibility defect level) is not exactly as we had intended. @dorseypa1 is going to fix that field by the end of this sprint (next Wednesday) but for now will hand off dataset and the team will get it into Domo (estimation in the meeting was EOD Friday 3/24/23). That will allow Ian access to it in the meantime. Does that all sound correct, @it-harrison ? cc // @shiragoodman

@sterkenburgsara Yes. Phil is going to keep the original dataset around for a while to assist with migrating to the new one.

**Comment by sterkenburgsara** [comment]
@it-harrison I am still seeing a 900-something number for the pratice-area card. Can you use the updated dataset to re-write that query per the tasks in this ticket? Still looks like the wrong count. 

@shiragoodman I added some text-only cards to section off the page - can you give me your feedback on that? Also, does first card (re: days to close) include launch-blocking tickets?

**Comment by shiragoodman** [comment]
Looking now. The Days to close card should include everything - launch-blocking and non-launch-blocking. When the card was created, we weren't pulling in the lb label so there was no way to single out lb or non lb.

**Comment by shiragoodman** [comment]
if by text-only cards you mean this:
<img width="490" alt="image" src="https://user-images.githubusercontent.com/71330581/228350983-7137d4c8-a59b-43ab-b00f-706f16ecd8ed.png">
then yes, I like them. They help frame what it is that we're looking at.

**Comment by shiragoodman** [comment]
Still waiting on Analytics and Insights to QA changes to the data set. As such, this ticket will move into the next sprint.

**Comment by michelle-dooley** [comment]
@shiragoodman and @it-harrison - Status update:  We are working on QA now and plan to have this completed by end of week.  cc: @dorseypa1

**Comment by michelle-dooley** [comment]
Also I noticed a few that have 508-defect-levels (0-4) but do not have the '508/Accessibility' label.  How should we manage those?  Seems like if you have a defect level you should also have the 508/Access label.  
![image](https://user-images.githubusercontent.com/93597659/228579942-3b675eea-ba6a-4752-9c2b-80a6323935da.png)

**Comment by shiragoodman** [comment]
@michelle-dooley thank you! Good catch. We can do a gh query to fix those tickets. I'll follow up.

**Comment by shiragoodman** [comment]
@michelle-dooley I'm seeing this message when I try and access the table/data set:
There was an error loading this DataSet. Please return to Data Center and try again later.

What is the name of the new data set so we can locate it through domo?

**Comment by shiragoodman** [comment]
@michelle-dooley thank you! This looks great. I think it should be much easier for us now to build cards as we no longer have a field with 1:many values. 

I'm assuming we'll need to update our cards to use the new data set before the old one can be deleted/archived. @it-harrison please take a look at the new data set compared to the[ old one. ](https://va-gov.domo.com/datasources/0375917b-befc-4057-b3f9-15eba2c78f2e/details/data/table) You can see in the old one that the ticket_number field has duplicate values, which is why our counts were always off.

Let me know an ETA/LOE for getting all of the cards recreated with the new data set. I'd advise not deleting the old cards until the new card is created and verified.

Thanks again, Michelle!

**Comment by michelle-dooley** [comment]
@shiragoodman yep update cards based on new dataset.  If you have any questions or need assistance while doing so just reach out.

**Comment by shiragoodman** [comment]
@michelle-dooley all our cards are updated with the new data set. Please feel free to archive/delete the old data set.

**Comment by shiragoodman** [comment]
@it-harrison please make sure to delete the outdated cards that have become irrelevant with the introduction of your new drill down card. After that, we can close this ticket.
